dio/lib/src/dio_error_extractor.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+/// Extracts the inner exception and stacktrace from [DioError]
+class DioErrorExtractor extends ExceptionCauseExtractor<DioError> {
+  @override
+  ExceptionCause? cause(DioError error) {
+    if (error.stackTrace == null) {
+      return null;
+    }
+    return ExceptionCause(
+      error.error ?? 'DioError inner stacktrace',
```

Comment:
I am not sure if this is correct, if `error.error` is null, that means there's no cause and we should bail, or does an inner sentry exception with `error.stackTrace` specifically still makes sense?
@ueman WDYT?

Reply:
Nvm, that was already like this, lets keep it as it is.

Reply:
It was like this before.
But I believe `error.error` is never actually `null`, it's just that the property is nullable. So this is basically just to be extra sure that we don't throw here.